src/patternfly/components/Nav/nav.scss

```diff
@@ -169,37 +204,75 @@
   --#{$nav}__item--RowGap--row-offset: calc(var(--#{$nav}__list--RowGap) / 2 * -1); // row gap offset calcs 1/2 height of row gap
   --#{$nav}__link--clickable-inset--Block: var(--#{$nav}__item--RowGap--row-offset); // set link's clickable area offset based on 1/2 row gap
   --#{$nav}__button--RowGap--row-offset: calc(var(--#{$nav}__item--RowGap) * -1); // match extra bottom paddings applied to expanded nav_links
+
+  visibility: visible;
 }
 
 // - Nav list
 .#{$nav}__list {
   row-gap: var(--#{$nav}__list--RowGap);
   column-gap: var(--#{$nav}__list--ColumnGap);
+  min-height: 0;
+  opacity: 1;
+  transition-timing-function: var(--#{$nav}__list--TransitionTimingFunction--opacity);
+  transition-duration: var(--#{$nav}__list--TransitionDuration--opacity);
+  transition-property: opacity;
+  transition-behavior: allow-discrete;
+
+  @starting-style {
+    opacity: 0;
+  }
 }
 
 // - Nav subnav
 .#{$nav}__subnav {
   --#{$nav}__list--RowGap: var(--#{$nav}__subnav--RowGap); // this value is passed to --#{$nav}__item--RowGap--row-offset and updates clickable area based on value passed
 
+  grid-template-rows: 1fr;
+  min-height: 0;
+  padding-block-start: 0;
+  padding-block-end: 0;
   padding-inline-start: var(--#{$nav}__subnav--PaddingInlineStart); // indent nested lists
+  overflow-y: clip;
+  transition-timing-function: var(--#{$nav}__subnav--TransitionTimingFunction--grid-template-rows);
+  transition-duration: var(--#{$nav}__subnav--TransitionDuration--grid-template-rows);
+  transition-property: grid-template-rows, padding-block-start;
+  transition-behavior: allow-discrete; 
+  
+  @starting-style {
+    grid-template-rows: 0fr;
+  }
```

Comment:
I don't think you need this actually? It starts out as 0fr already https://github.com/patternfly/patternfly/blob/b635f763cec40fb2cf4ae5f84ec04f31a83fde6d/src/patternfly/components/Nav/nav.scss#L246-L250

Reply:
Are you saying we don't need the `@starting-style` block? That allows the animation to run on page load (same with the next comment, if that's what you meant.)

Reply:
Ah! I didn't know those were for firing on page load. I'll re-review 👍